rundeckapp/grails-app/services/rundeck/services/asyncimport/AsyncImportService.groovy

```diff
@@ -0,0 +1,927 @@
+            return it
+        })
+
+        Path pathToRundeckInternalProject = Files.list(Paths.get(modelProjectHost.toString()))
+                .filter { it ->
+                    it.fileName.toString().startsWith("rundeck-")
+                }.collect(Collectors.toList())[0]
+
+        List<Path> filepathsToRemove = Files.list(pathToRundeckInternalProject).filter {
+            it -> it.fileName.toString() != "jobs"
+        }.collect(Collectors.toList())
+
+        filepathsToRemove.forEach {
+            it ->
+                {
+                    File file = new File(it.toString())
+                    if (file.isDirectory()) {
+                        deleteNonEmptyDir(file.toString())
+                    } else {
+                        Files.delete(it)
+                    }
+                }
+        }
+
+        asyncImportStatusFileUpdater(new AsyncImportStatusDTO(projectName, milestoneNumber).with {
+            it.lastUpdate = "Milestone 1 completed, calling Milestone 2 in process..."
+            return it
+        })
+
+        if( importExecutions && executionsDirFound ){
+            projectService.beginAsyncImportMilestone(
+                    projectName,
+                    authContext,
+                    project,
+                    AsyncImportMilestone.M2_DISTRIBUTION.milestoneNumber
+            )
+        }else{
+            // End the async import
+            if( Files.exists(Paths.get(destDir)) ) deleteNonEmptyDir(destDir)
+            if( Files.exists(Paths.get(scopedWorkingDir)) ) deleteNonEmptyDir(scopedWorkingDir)
+            asyncImportStatusFileUpdater(new AsyncImportStatusDTO(projectName, milestoneNumber).with {
+                it.milestone = AsyncImportMilestone.ASYNC_IMPORT_COMPLETED.name
+                it.lastUpdate = "All Executions uploaded, async import ended. Please check the target project."
+                return it
+            })
+        }
+
+        return importResult
+    }
+
+
+    /**
+     * Async import process Milestone 2: Files Distribution.
+     * This method is asynchronous and only will be triggered by the completion of milestone 1 through an event emission,
+     * it:
+     * 1) Creates the distributed executions dir in working directory for project
+     * 2) Extracts the project copy in temp for a given project and list each execution of the project, then moves the executions
+     * and its corresponding files to a bundle, which will have a maximum of 1000 executions per bundle or by the number passed to the prop
+     * "rundeck.asyncImportConfig.maxDistributedExecutions". If the max is reached a next bundle will be created
+     * 2) Removes the copy of the uploaded project in /tmp path
+     * 3) Calls for the Milestone 3.
+     *
+     *
+     * @param projectName - String
+     * @param authContext - AuthContext
+     * @param project - IRundeckProject
+     */
+    @Subscriber(AsyncImportEvents.ASYNC_IMPORT_EVENT_MILESTONE_2)
+    def beginMilestone2(
```

Comment:
check if this method can be separated into multiple methods for better reading and easier testing